Specification.md

```diff
@@ -0,0 +1,87 @@
+
+## Requirements
+
+### Licensing
+
+Xapps need to be licensed on a [FOSS license](https://opensource.org/licenses).
+
+### Operating System
+
+Xapps need to be able to run in a Linux environment. The support of other operating systems is optional.
+
+### Graphical Desktop Application
+
+This project targets graphical applications which run on desktop systems and dependencies of them. E.g. console applications, kernel-modules and system-services cannot be xapps.
+
+### Usage of standards
+
+In order to be desktop-environment agnostic, the compliance to the following [freedesktop.org](https://www.freedesktop.org) standards is mandatory to be listed as an xapp:
+
+- [Xdg Autostart](https://www.freedesktop.org/wiki/Specifications/autostart-spec/)
+- [Xdg Base Directories](https://www.freedesktop.org/wiki/Specifications/basedir-spec/)
+- [Xdg Desktop Entry](https://www.freedesktop.org/wiki/Specifications/desktop-entry-spec/)
+- [Xdg Desktop Menus](https://www.freedesktop.org/wiki/Specifications/menu-spec/)
+- [File Manager DBUS interface](https://www.freedesktop.org/wiki/Specifications/file-manager-interface/)
+- [File URI Specification](https://www.freedesktop.org/wiki/Specifications/file-uri-spec/)
+- [Free Media Player Specification](https://www.freedesktop.org/wiki/Specifications/free-media-player-specs/)
+- [Free Media Player Remote Interface Specification](https://www.freedesktop.org/wiki/Specifications/mpris-spec/)
+- [Xdg Icon Theme](https://www.freedesktop.org/wiki/Specifications/icon-theme-spec/)
+- [Xdg Icon Naming](https://www.freedesktop.org/wiki/Specifications/icon-naming-spec/)
+- [Shared mime Database](https://www.freedesktop.org/wiki/Specifications/shared-mime-info-spec/)
+- [Startup Notification](https://www.freedesktop.org/wiki/Specifications/startup-notification-spec/)
+- [Trashcan support](https://www.freedesktop.org/wiki/Specifications/trash-spec/)
+- [Bookmark Interchange Format](https://pyxml.sourceforge.net/topics/xbel/)
+
+TBD: Not sure on these:
+- [MIME Applications Associations](https://www.freedesktop.org/wiki/Specifications/mime-apps-spec/)
+- [XSettings](https://www.freedesktop.org/wiki/Specifications/xsettings-spec/)
+- [Cursor Conventions](https://www.freedesktop.org/wiki/Specifications/cursor-spec/)
+- [Sound Theme](https://www.freedesktop.org/wiki/Specifications/sound-theme-spec/)
+
+### Window decorations
+
+Applications which are part of the xapp project should support server side window decorations and use them by default. Xapps are free to optionally as well support client side decorations [^1].
+
+### Menus
+
+If menus are required for a xapps, traditional menu bars should be used by default. It is fine to optionally support e.g. hamburger menus [^2].
+
+TBD: Do we want to force a default on this?
```

Comment:
The [original Xapp guide](https://linuxmint-developer-guide.readthedocs.io/en/latest/xapps.html) requested traditional menus, so I added it as well here ... though not sure if we still want that.

Reply:
I'm not sure I agree with the idea of mandating support for traditional menubars, as this may exclude applications which would not otherwise benefit from them, create unnecessary development / maintenance overhead (if needing to maintain multiple implementations for said functionality), or run counter to any desired HIG they wish to follow (which doesn't necessarily mandate any given theming).

Each application under Xapps should ideally be able to consider the usefulness and/or relevance of the menubar. If the application is written in something like Kirigami + Qt as an example, the application may choose alternative navigation methods such as a toolbar or a navigation toolbar.

- https://develop.kde.org/hig/components/navigation/menubar/
- https://develop.kde.org/hig/components/navigation/toolbar/
- https://develop.kde.org/hig/components/navigation/navigationtabbar/

Applications written in GTK4 (not necessarily with libadwaita) could theoretically use GtkPopoverMenuBar, but given they would be using GtkHeaderBar, it's more likely they would just be putting controls in that unless their application _needs_ more complex navigation (hence they should be considering the usefulness of menubar rather than it being forced).

Reply:
Agreed; I don't think we should mandate menubars be used.  I think they're appropriate in some cases (like OpenOffice, where I use the menus ever time I use the app), but not so much in others (I have Firefox's menubar hidden, and I can't remember the last time I used it).

And similar to my comment about supporting the various standards, I want to make sure that prospective xapp authors don't read this and think that they're required to have a menubar, even if they don't need one.

Reply:
I don't have a strong opinion on this one, though I tend to agree with @JoshStrobl and @kelnos for it. It would be fine for me to just drop the menubar paragraph ... though let's wait for the opinion of @clefebvre

Reply:
Alright, dropped the paragraph "Menus": d0503c71f1a4ac227eb3e70da6496d6e716c2874 - closing this thread